### PR TITLE
APP-6785: Remove local control page (RC)

### PR DESCRIPTION
**Pull request by ethanlookpotts**
We are removing RC from the RDK - users are now expected to use app.viam.com to control their machine.

> [!NOTE]
> We don't expect workflows to pass until the base PR has been merged.

1. #4523
2. 👉🏼 #4520 

## Change log

- Remove `web/frontend`
- Remove unused code in `web/`
- Remove unused `e2e` script
- Remove web-related targets from `Makefile`
- Remove web-related docs from `README`

## Review requests

- Please review `robot/web/web.go` thoroughly - I aimed to remove anything related to serving the web app, but I might've trimmed too much or missed some code. Let me know!
- Anyone missing from the reviewer list?

### CODE REVIEW

File: robot/web/web.go [hunk 1]
```diff
@@ -839,11 +686,9 @@ func (svc *webService) initHTTPServer(listenerTCPAddr *net.TCPAddr, options webo
 }
 
 // Initialize multiplexer between http handlers.
-func (svc *webService) initMux(options weboptions.Options) (*goji.Mux, error) {
+func (svc *webService) initMux(options weboptions.Options) *goji.Mux {
 	mux := goji.NewMux()
-	if err := svc.installWeb(mux, svc.r, options); err != nil {
-		return nil, err
-	}
+	mux.Handle(pat.New("/"), http.NotFoundHandler())
```
Comment (ethanlookpotts) [inline robot/web/web.go:691]: @Otterverse see here - not found for `/`. How do health checks work for the agent?

Comment (Otterverse) [reply robot/web/web.go:691]: It looks for a 2XX status code on whatever URL viam-server gives as it's "serving" line during startup. Ex:

```
11/5/2024, 12:35:22 PM info rdk web/web.go:599 serving url https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 alt_url https://0.0.0.0:8080

11/5/2024, 12:35:22 PM info viam-agent viamserver/viamserver.go:172 healthcheck URLs: https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 https://localhost:8080

11/5/2024, 12:35:22 PM info viam-agent viamserver/viamserver.go:173 viam-server started

```

Basically agent sees that first line and parses it, and reports WHAT it parsed in that second line. Then it queries that every minute, looking for a 2XX http response (it doesn't care about the response body.)

Comment (ethanlookpotts) [reply robot/web/web.go]: Awesome! I added a 200 status and "healthy" message here: [596b486](https://github.com/viamrobotics/rdk/pull/4520/commits/596b4860221b75452c81bc2ae55a2d6270fe619f)

@Otterverse are you able to verify that this works if I generate an AppImage?

Comment (Otterverse) [reply robot/web/web.go]: Agent doesn't use appimages. Those are the "old" method. Just do a static binary `make server-static` and then you can test it yourself. You can copy the built file to your device (running agent) and set the pin_url for viam-server to file:///path/to/local/viam-server to let it run the new version. (You can also use a remote/public URL if you find that easier.)

Comment (Otterverse) [reply robot/web/web.go]: I can try the same now too... will clone your branch and give it a whirl.

Comment (Otterverse) [reply robot/web/web.go]: Okay, running a build from this branch, looks good!

```
2024-11-05T20:19:31.030Z	INFO	rdk	web/web.go:449	serving	{"url":"https://artoo-main.p55vcc4g4x.local.viam.cloud:8080","alt_url":"https://0.0.0.0:8080"}
2024-11-05T20:19:31.031Z	INFO	viam-agent	viamserver/viamserver.go:172	healthcheck URLs: https://artoo-main.p55vcc4g4x.local.viam.cloud:8080 https://localhost:8080

...

2024-11-05T20:20:34.390Z	DEBUG	viam-agent	viamserver/viamserver.go:290	healthcheck for viam-server is good
2024-11-05T20:20:34.391Z	DEBUG	viam-agent	agent/manager.go:233	Subsystem healthcheck succeeded for viam-server
```

Comment (ethanlookpotts) [reply robot/web/web.go]: Sorry for the delay here - thank you for taking a look! I am also going to try going through these steps for my understanding. Once I've confirmed that things are working as expected I'll move forward with the base PR and put this one up for explicit review.

Comment (ethanlookpotts) [reply robot/web/web.go]: Sweet! I was able to verify this as well and everything is behaving as expected.

**Comment by dgottlieb** [comment]
> We are removing RC from the RDK - users are now expected to use app.viam.com to control their machine.

So this breaks using `localhost:8080` ? Is there some alternative workflow for robots using a "local FS" config?

**Comment by benjirewis** [comment]
Was also looking @dgottlieb ; assume this is relevant https://viam.atlassian.net/wiki/spaces/ENG/pages/829587457/Local+control+page

**Comment by ethanlookpotts** [comment]
Correct @dgottlieb, this breaks using `localhost:8080`. The [confluence page](https://viam.atlassian.net/wiki/spaces/ENG/pages/829587457/Local+control+page) that @benjirewis linked is how you can connect with cloud signaling, but not sure if it works in the "local FS" case. Trying now.

**Comment by dgottlieb** [comment]
Thanks for that document link! But I'm not sure if that helps me? If I wanted to avoid using app, I don't think running a local app instance is the direction I was hoping for.

**Comment by ethanlookpotts** [comment]
> If I wanted to avoid using app, I don't think running a local app instance is the direction I was hoping for.

It doesn't run the whole local app instance - just the control tab, just like the page on `localhost:8080` used to be. The code does live in the app repo, but you don't run the whole app. Acknowledged that it doesn't help OSS users, only our employees at the moment.

File: robot/web/web.go [hunk 2]
```diff
@@ -55,120 +50,6 @@ var API = resource.APINamespaceRDKInternal.WithServiceType(SubtypeName)
...
-type AppTemplateData struct {
-	WebRTCEnabled          bool                   `json:"webrtc_enabled"`
-	WebRTCSignalingAddress string                 `json:"webrtc_signaling_address"`
-	Env                    string                 `json:"env"`
-	Host                   string                 `json:"host"`
-	StaticHost             string                 `json:"static_host"`
-	SupportedAuthTypes     []string               `json:"supported_auth_types"`
-	AuthEntity             string                 `json:"auth_entity"`
-	BakedAuth              map[string]interface{} `json:"baked_auth"`
-}
```
Comment (randhid) [inline robot/web/web.go:105]: can new local app handle no credentials passed, insecure, like the one you're deleting did?

I can not run rdk without credentials and use the sdks to connect to it with an insecure option, if this possible in the new world where we build "local" rc?

Comment (zaporter-work) [reply robot/web/web.go:105]: Insecure connections haven't been built into the svelte-sdk. I can add them easily if it would be useful to you

Comment (ethanlookpotts) [reply robot/web/web.go:105]: This is what @dgottlieb is looking for as well - we need a small TS SDK change in order to provide this from the `svelte-sdk` (aka the new local control page). Opened [here](https://github.com/viamrobotics/viam-typescript-sdk/pull/411), and will also open the corresponding change against the `svelte-sdk`.

**Comment by ethanlookpotts** [comment]
This is still ready for review but I will wait to merge it until we have the solution for the new local control page connecting insecurely to a local machine running a local config. (This PR will not make the next `viam-server` release.)

- [ ] https://github.com/viamrobotics/viam-typescript-sdk/pull/411
- [ ] https://github.com/viamrobotics/app/pull/6824

File: web/app_fs.go [hunk 3]
```diff
@@ -1,9 +0,0 @@
-// Package web contains the root of a web server.
-package web
-
-import "embed"
-
-// AppFS is the embedded FS to control a robot with.
-//
-//go:embed runtime-shared
-var AppFS embed.FS
```
[the remark this answers is not in the document]
Comment (ethanlookpotts) [reply web/app_fs.go:9]: Thanks for the callout! Done [here](https://github.com/viamrobotics/app/pull/6841)!

**Comment by ethanlookpotts** [comment]
> I saw the TS changes were merged. Would it be helpful for me to test run a local config with the new proposed flow?

@dgottlieb Not quite yet, but I will soon! We need the TS SDK to be bumped in App (happening this afternoon) and then we can move forward with https://github.com/viamrobotics/app/pull/6824. Then I will update the instructions and ask you to test the new workflow for me! Thank you!

**Review by randhid — APPROVED** [review]
Let me know if you need someone to manually test the local app build with insecure credentials when that is ready.

**Comment by ethanlookpotts** [comment]
I've verified with @dgottlieb and @randhid that the new local control page in App works for their needs.

I'm going to go ahead and merge this tomorrow so it makes it into the next RDK release. Please review if needed!

**Comment by benjirewis** [comment]
Taking myself off review here. Makes sense/LGTM though.

**Review by zaporter-work — APPROVED** [review]

**Pull request merged**